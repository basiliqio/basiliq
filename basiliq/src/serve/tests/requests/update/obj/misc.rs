use super::*;

lazy_static::lazy_static! {
    pub static ref UPDATE_BODY: serde_json::Value =
        json!({
            "data": json!({
                "type": "public__peoples",
                "id": PEOPLES_IDS[0],
                "attributes": json!({
                    "age": 100
                })
            })
        });
}

crate::run_test_request!(
    sorting_local,
    Method::PATCH,
    format!("/public__peoples/{}?sort=first-name", PEOPLES_IDS[0]),
    200,
    UPDATE_BODY.clone()
);

crate::run_test_request!(
    sorting_local_multi,
    Method::PATCH,
    format!(
        "/public__peoples/{}?sort=first-name,-last-name",
        PEOPLES_IDS[0]
    ),
    200,
    UPDATE_BODY.clone()
);

crate::run_test_request!(
    sorting_remote,
    Method::PATCH,
    format!(
        "/public__peoples/{}?sort=-public__favorite_color.color",
        PEOPLES_IDS[0]
    ),
    200,
    UPDATE_BODY.clone()
);

crate::run_test_request!(
    sorting_remote_multi_rel,
    Method::PATCH,
    format!(
        "/public__peoples/{}?sort=public__articles.title",
        PEOPLES_IDS[0]
    ),
    403,
    UPDATE_BODY.clone()
);

crate::run_test_request!(
    sorting_and_include_and_sparse,
    Method::PATCH,
    format!("/public__peoples/{}?sort=public__favorite_color.color&include=public__favorite_color&fields[public__favorite_color]=", PEOPLES_IDS[0]),
    200,
    UPDATE_BODY.clone()
);

crate::run_test_request!(
    add_relationship_m2m,
    Method::PATCH,
    format!("/public__peoples/{}", PEOPLES_IDS[0]),
    403,
    json!({
        "data": json!({
            "type": "public__peoples",
            "id": PEOPLES_IDS[0],
            "attributes": json!({
                "age": 100
            }),
            "relationships": json!({
                "public__articles": json!({
                    "data": json!({
                        "id": ARTICLES_IDS[2],
                        "type": "public__articles"
                    })
                })
            })
        })
    })
);

crate::run_test_request!(
    add_relationship_m2o,
    Method::PATCH,
    format!("/public__peoples/{}", PEOPLES_IDS[0]),
    403,
    json!({
        "data": json!({
            "type": "public__peoples",
            "id": PEOPLES_IDS[0],
            "attributes": json!({
                "age": 100
            }),
            "relationships": json!({
                "public__people-article": json!({
                    "data": json!({
                        "id": PEOPLE_ARTICLE_ID[2],
                        "type": "public__people-article"
                    })
                })
            })
        })
    })
);
