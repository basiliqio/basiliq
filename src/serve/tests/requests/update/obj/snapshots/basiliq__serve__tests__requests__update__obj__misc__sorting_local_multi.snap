---
source: basiliq/src/serve/tests/requests/update/obj/misc.rs
expression: res

---
{
  "jsonapi": {
    "version": "1.0"
  },
  "data": {
    "type": "public__peoples",
    "id": "[uuid]",
    "attributes": {
      "age": 100,
      "first-name": "Francis",
      "gender": "M",
      "last-name": "Le Roy",
      "twitter": null
    }
  }
}
